test-refined/shared/src/main/scala/zio/test/refined/CollectionInstances.scala

```diff
@@ -2,73 +2,99 @@ package zio.test.refined
 
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.collection.{NonEmpty, Size}
-import zio.test.Gen
+import zio.random.Random
 import zio.test.magnolia.DeriveGen
+import zio.test.{Gen, Sized}
 import zio.{Chunk, NonEmptyChunk}
 
 object collection extends CollectionInstances
 
 trait CollectionInstances {
+
+  def nonEmptyChunkRefinedGen[R <: Random with Sized, C, T](implicit
+    genT: Gen[R, T]
+  ): Gen[R, Refined[NonEmptyChunk[T], NonEmpty]] =
+    Gen.chunkOf1(genT).map(Refined.unsafeApply)
+
+  def nonEmptyListRefinedGen[R <: Random with Sized, T](implicit
+    genT: Gen[R, T]
+  ): Gen[R, Refined[List[T], NonEmpty]] = Gen.listOf1(genT).map(Refined.unsafeApply)
+
+  def nonEmptyVectorRefinedGen[R <: Random with Sized, T](implicit
+    genT: Gen[R, T]
+  ): Gen[R, Refined[Vector[T], NonEmpty]] = Gen.vectorOf1(genT).map(Refined.unsafeApply)
+
+  def sizedChunkRefinedGen[R <: Random with Sized, T, P](implicit
+    genT: Gen[R, T],
+    sizeGen: Gen[Random, Int Refined P]
+  ): Gen[R, Refined[Chunk[T], Size[P]]] =
+    sizedChunkGen(genT, sizeGen)
+      .map(r => Refined.unsafeApply(r))
+
+  def listSizeRefinedGen[R <: Random with Sized, T, P](implicit
+    genT: Gen[R, T],
+    sizeGen: Gen[Random, Int Refined P]
+  ): Gen[R, Refined[List[T], Size[P]]] =
+    sizedChunkGen(genT, sizeGen)
```

Comment:
I would do `sizeGen.flatMap(n => listOfN(n.value)(genT))` and similarly below to avoid the unnecessary conversion between collection types.

Reply:
But this does not extract their logic into the new method.

Reply:
Maybe I'm not understanding. It seems like you want to construct three things in this and the methods below:
1. Chunks with size that confirms to a refinement type
2. Lists with size that conforms to a refinement type
3. Vectors with size that confirms to a refinement type

Right now you're implementing `sizedChunkGen` which generates (1) and then you are using the `toList` and `toVector` operators to generate (2) and (3). From a code re-use perspective that uses slightly less code since you are using `sizedChunkGen` everywhere.

However, it means that the generators (2) and (3) above are unnecessarily inefficient because they first build a `Chunk` and then tear it down and rebuild it into a `List` or `Vector`. Furthermore, the logic in `sizedChunkGen` is fairly trivial, since it basically just combines two existing operators in the most straightforward way possible. So the gains to this versus just implementing (2) and (3) directly in terms of `listOfN` and `vectorOfN` is minimal.

Does that make sense?

Reply:
Oh, I know how to do it.